client/src/app/components/Autocomplete.tsx

```diff
@@ -256,9 +331,48 @@ export const Autocomplete: React.FC<IAutocompleteProps> = ({
     </div>
   );
 
-  const menu = (
-    <Menu ref={menuRef} onKeyDown={handleMenuOnKeyDown} isScrollable>
-      <MenuContent>
+  const renderMenuItems = () => {
+    if (isGrouped && groupedOptions) {
+      return Object.entries(groupedOptions).map(([groupName, groupOptions]) => (
+        <React.Fragment key={groupName}>
+          <MenuGroup label={groupName}>
```

Comment:
[Looking at the PF example](https://www.patternfly.org/components/menus/menu#titled-groups-of-items) a `MenuGroup` without a label is effectively the same as no group.
Perhaps we could avoid branching here by treating `isGrouped== false` as a group with no name (`groupName == undefined`)